manager-bundle/src/ContaoManager/Plugin.php

```diff
@@ -291,32 +292,15 @@ private function addDefaultPdoDriverOptions(array $extensionConfigs, ContainerBu
             return $extensionConfigs;
         }
 
-        $driver = null;
-        $url = null;
-
-        foreach ($extensionConfigs as $extensionConfig) {
-            // Do not add PDO options if custom options have been defined
-            // Since this is merged recursively, we don't need to check other configs
-            if (isset($extensionConfig['dbal']['connections']['default']['options'][\PDO::MYSQL_ATTR_MULTI_STATEMENTS])) {
-                return $extensionConfigs;
-            }
-
-            if (isset($extensionConfig['dbal']['connections']['default']['driver'])) {
-                $driver = $extensionConfig['dbal']['connections']['default']['driver'];
-            }
+        [$driver, $options] = $this->parseDbalDriverAndOptions($extensionConfigs, $container);
 
-            if (isset($extensionConfig['dbal']['connections']['default']['url'])) {
-                $url = $container->resolveEnvPlaceholders($extensionConfig['dbal']['connections']['default']['url'], true);
-            }
-        }
-
-        // If URL is set it overrides the driver option
-        if (null !== $url) {
-            $driver = str_replace('-', '_', parse_url($url, PHP_URL_SCHEME));
+        // Do not add PDO options if custom options have been defined
+        if (isset($options[\PDO::MYSQL_ATTR_MULTI_STATEMENTS])) {
+            return $extensionConfigs;
         }
 
         // Do not add PDO options if the selected driver is not mysql
-        if (null !== $driver && !\in_array($driver, ['pdo_mysql', 'mysql', 'mysql2'], true)) {
+        if (null !== $driver && 'mysql' !== $driver) {
```

Comment:
Why did you remove the array? Can `$driver` no longer be `pdo_mysql` or `mysql2`?

Reply:
Ah, I see. You are normalizing it in the `parseDbalDriverAndOptions()` method. @ausi Is this safe?

Reply:
IMO yes.